inc/providers/class-core-sitemaps-taxonomies.php

```diff
@@ -77,21 +77,37 @@ public function get_url_list( $page_num, $taxonomy = '' ) {
 		// Offset by how many terms should be included in previous pages.
 		$offset = ( $page_num - 1 ) * core_sitemaps_get_max_urls( $this->object_type );
 
-		$args = array(
-			'fields'                 => 'ids',
-			'taxonomy'               => $taxonomy,
-			'orderby'                => 'term_order',
-			'number'                 => core_sitemaps_get_max_urls( $this->object_type ),
-			'offset'                 => $offset,
-			'hide_empty'             => true,
-
-			/*
-			 * Limits aren't included in queries when hierarchical is set to true (by default).
-			 *
-			 * @link: https://github.com/WordPress/WordPress/blob/5.3/wp-includes/class-wp-term-query.php#L558-L567
-			 */
-			'hierarchical'           => false,
-			'update_term_meta_cache' => false,
+		/**
+		 * Filters the taxonomy terms query arguments.
+		 *
+		 * Allows modification of the taxonomy query arguments before querying.
+		 *
+		 * @see WP_Term_Query for a full list of arguments
+		 *
+		 * @since 5.5.0
+		 *
+		 * @param array  $args     An array of WP_Term_Query arguments.
+		 * @param string $taxonomy The taxonomy string.
+		 */
+		$args = apply_filters(
+			'core_sitemaps_taxonomy_terms_query_args',
+			array(
+				'fields'                 => 'ids',
+				'taxonomy'               => $taxonomy,
+				'orderby'                => 'term_order',
+				'number'                 => core_sitemaps_get_max_urls( $this->object_type ),
+				'offset'                 => $offset,
+				'hide_empty'             => true,
+
+				/*
+				* Limits aren't included in queries when hierarchical is set to true (by default).
+				*
+				* @link: https://github.com/WordPress/WordPress/blob/5.3/wp-includes/class-wp-term-query.php#L558-L567
+				*/
+				'hierarchical'           => false,
+				'update_term_meta_cache' => false,
+			),
+			$taxonomy
```

Comment:
General issue here: If we add a filter for the URL list query args, we should also add a filter for the "max num pages" query args like you did for posts.

I wonder though if we could simplify this, since having to add two filters for each of these things is not a great developer experience. We should be able to instead restructure the code in both the posts and taxonomies provider to use the same array of query arguments, and then just modify the `fields` / `count` / `no_found_rows` arguments _after_ filtering, based on whether we're getting items or counting items.

This is already the case for the users provider, and I think it would be much cleaner if we did this throughout. We can use the same type of `WP_Query` / `WP_Term_Query` instances respectively. How about we introduce a `get_public_terms( $taxonomy )` method here that returns the common "base" query args? We could then have a single `core_sitemaps_taxonomies_query_args` filter there to cover both areas.

Reply:
`get_public_terms_query` perhaps? More in line with `\Core_Sitemaps_Users::get_public_post_authors_query`. Or ideally use the same method name for all providers, i.e. `get_items_query_args`.

Also, should those methods be protected?

Reply:
@swissspidy should the $args array for get_url_list and max_num_pages be the same?

Reply:
I would vote for the second... it provides more consistence (like `max_num_pages`) and the differentiation comes with the class context.

Reply:
> should the `$args` array for `get_url_list` and `max_num_pages` be the same?

I think that's what @felixarntz is proposing, yes.

However, `max_num_pages` would then modify the `$args` _afterwards_ to set `fields=>'ids', 'no_found_rows'=>false`, etc.

Example:

```php

public function get_items_query_args() {
 ...
 return apply_filters( '', $args );
}

public function max_num_pages() {
  $args = $this->get_items_query_args();
  $args['fields'] => 'ids'

  ...
}
```

Pros:

Less duplication

Cons:

Not possible to completely filter all args used in `max_num_pages`

Reply:
@swissspidy we should also rename `max_num_pages` to `get_max_num_pages` then?

Reply:
Yeah I think that makes sense.

Reply:
should I rename them as part of this MR?

Reply:
it's not that easy to encapsulate the `get_items_query_args` in a generic/abstract function (like `max_num_pages`), because the queries have different params, like for example `offset` for `WP_Term_Query` and `paged` for `WP_Query`.

Reply:
Oh, I think each provider would implement this on their own, no? It wouldn't live in the abstract provider class.

Reply:
I don't think we should abstract the methods into like the base provider class, I think it makes sense for these methods to remain in the respective specific provider (I think protected methods make sense because they're implementation details for the respective provider). I don't necessarily think we need to use abstract method names because these methods are specific. How about `get_posts_query_args()` and `get_terms_query_args()`?

Reply:
@swissspidy 
> Not possible to completely filter all args used in `max_num_pages`

Would this actually be a problem though? I'd argue not being able to filter arguments like `count` is actually cleaner because changing this would likely mess things up anyway by e.g. changing the return type of the query. I'd also say we shouldn't include any pagination arguments in the query, these should also get applied after filtering. What these filters should be used for is modifying the query details for what is queried overall, but not how we paginate or count items.